test/index.js

```diff
@@ -183,3 +183,171 @@ describe('metalsmith-serve with custom indexFile', function(){
 
 });
 
+
+// not_found file serving and redirects
+describe('metalsmith-serve', function() {
+
+  var metalsmith;
+  var servePlugin;
+
+  before(function(done) {
+    metalsmith = Metalsmith("test/fixtures/site");
+
+    servePlugin = serve({
+      verbose: false,
+      "port": port,
+      "not_found": "/404.html",
```

Comment:
Would it make sense to have something like:
```
{
  verbose: false,
  ...
  "http_codes": {
    404: "/404.html"
  },
  ...
}
```

and then simply use the error codes to pull out the custom pages? It would make it a little more generic and go beyond just custom 404, but hey, why not? (`if (options.http_codes[err.status])` in `lib/index.js` line 33)